4.2_exact_divisor.py

```diff
@@ -0,0 +1,8 @@
+def exact_divisor(number):
+    """Finds the exact divisors of the number
+    Returns a list"""
+    exact_divisors_list = [i for i in range(1, number+1) if number % i == 0]
+    return exact_divisors_list
+
+
+print(exact_divisor(int(input('Give me a number then i find its exact divisors: '))))
```

Comment:
- kodlarınız çalışıyor tebrikler.

asağıda degisik bir cözüm var. ama bu fonksiyonu anlamanız için yield kullanımını internetten incelemeniz gerekiyor. yield çok kullanılmasada bilgi sahibi olmakta fayda var.
ama özellikle bu sorunun çözümünde değer olarak 1.000.000.000.000.000 gibi büyük bir sayı girdiğinizde aşağıdaki kodlar ile sizin çözümünüzdeki kodların şaşırtıcı derecede çözüm süresi farkına şahit olursunuz. çözüm süresinin kısalması yield ile fazla ilgili değil çözüm yönteminden kaynaklanıyor.

import math
def divisorGenerator(n):
large_divisors = []
for i in range(1, int(math.sqrt(n) + 1)):
if n % i == 0:
yield i
if i*i != n:
large_divisors.append(n / i)
for divisor in reversed(large_divisors):
yield divisor

print (list(divisorGenerator(100000000000000)))